modules/curl/7.69.1/patches/add_build_file.patch

```diff
@@ -0,0 +1,748 @@
+--- /dev/null
++++ BUILD.bazel	
+@@ -0,0 +1,745 @@
++# This is based on:
++#   https://raw.githubusercontent.com/googleapis/google-cloud-cpp/e3dfb90502e5f7b844f5dc7452bf8cd249353c57/bazel/curl.BUILD
++
++# Copyright 2018 Google LLC
++#
++# Licensed under the Apache License, Version 2.0 (the "License");
++# you may not use this file except in compliance with the License.
++# You may obtain a copy of the License at
++#
++#     https://www.apache.org/licenses/LICENSE-2.0
++#
++# Unless required by applicable law or agreed to in writing, software
++# distributed under the License is distributed on an "AS IS" BASIS,
++# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
++# See the License for the specific language governing permissions and
++# limitations under the License.
++
++# Description:
++#   curl is a tool for talking to web servers.
++
++licenses(["notice"])  # MIT/X derivative license
++
++exports_files(["COPYING"])
++
++config_setting(
++    name = "windows",
++    constraint_values = [
++        "@platforms//os:windows",
++        "@platforms//cpu:x86_64",
++    ],
++    visibility = ["//visibility:public"],
++)
++
++config_setting(
++    name = "macos",
++    constraint_values = [
++        "@platforms//os:macos",
++    ],
++    visibility = ["//visibility:public"],
++)
++
++# On Linux, libcurl needs to know, at compile time, the location for the
```

Comment:
The CA cert logic is problematic since it inspects the host in a `genrule`, which will not work correctly with remote execution. We could define a Starlark `string_flag` and require Linux users to set it explicitly.